diff -uraN asahi-scripts-20221220/initcpio/hooks/asahi sd-asahi-scripts/initcpio/hooks/asahi
--- asahi-scripts-20221220/initcpio/hooks/asahi	2022-12-20 08:29:20.000000000 +0000
+++ sd-asahi-scripts/initcpio/hooks/asahi	2022-12-22 22:29:59.609887504 +0000
@@ -50,6 +50,7 @@
 }
 
 run_latehook() {
+    [ -e /vendorfw ] || return
     msg ":: Asahi: Copying vendor firmware to tmpfs under root filesystem..."
     mkdir -p /new_root/lib/firmware/vendor
     mount -t tmpfs vendorfw /new_root/lib/firmware/vendor
diff -uraN asahi-scripts-20221220/update-m1n1 sd-asahi-scripts/update-m1n1
--- asahi-scripts-20221220/update-m1n1	2022-12-20 08:29:20.000000000 +0000
+++ sd-asahi-scripts/update-m1n1	2022-12-22 22:29:59.609887504 +0000
@@ -5,6 +5,8 @@
 
 [ -e /etc/default/update-m1n1 ] && . /etc/default/update-m1n1
 
+[ -n $M1N1_UPDATE_DISABLED ] && exit 0
+
 if [ -e "$(dirname "$0")"/functions.sh ]; then
     . "$(dirname "$0")"/functions.sh
 else
